airbyte-commons/src/main/java/io/airbyte/commons/logging/MaskedDataInterceptor.java

```diff
@@ -127,7 +127,7 @@ private String generatePattern(final Set<String> properties) {
     builder.append("(?i)"); // case insensitive
     builder.append("\"(");
     builder.append(properties.stream().collect(Collectors.joining("|")));
-    builder.append(")\"\\s*:\\s*\"?((\\\\\"|[^\",}])*)\"?");
+    builder.append(")\"\\s*:\\s*(\"(?:[^\"\\\\]|\\\\.)*\"|\\[[^]\\[]*]|\\d+)");
```

Comment:
nit: Feels like it start to make sense to add the not escaped regex as a comment for readability.